bootloader/src/pyi_utils.c

```diff
@@ -859,6 +891,7 @@ pyi_utils_create_child(const char *thisfile, const int argc, char *const argv[])
     /* Child code. */
     if (pid == 0) {
         /* Replace process by starting a new application. */
+        set_systemd_env();
```

Comment:
If this function is going to return a result code, you should at least check it and print a warning message on failure.

Other than that, looks good to me!